pkg/segment/results/segresults/segresults.go

```diff
@@ -416,6 +416,12 @@ func (sr *SearchResults) GetTotalCount() uint64 {
 	return sr.resultCount
 }
 
+func (sr *SearchResults) GetAggs() *structs.QueryAggregators {
+	sr.updateLock.Lock()
+	defer sr.updateLock.Unlock()
+	return sr.sAggs
+}
```

Comment:
this method is not giving you the mutex protection that you intend to do

Reply:
since `sAggs` is only used as readonly thing, while the other members of this struct have a requirement to be thread protected (since concurrent results addition happens)
we can remove the marshall/unmarshall and just have the lock/unlock piece as it was before